src/main/java/org/example/daos/AuthDao.java

```diff
@@ -0,0 +1,50 @@
+package org.example.daos;
+
+import org.example.exceptions.DatabaseConnectionException;
+import org.example.models.LoginRequest;
+import org.example.models.User;
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static org.example.util.PasswordEncoder.getPasswordEncoder;
+
+public class AuthDao {
+    private final int maxSetStringNumber = 3;
+
+    public User getUser(final LoginRequest loginRequest)
+            throws SQLException, DatabaseConnectionException {
+        try (Connection connection = DatabaseConnector.getConnection()) {
+            String query = "SELECT `email`, `password`, `roleId`"
+                    + "FROM `User`"
+                    + "WHERE `email` = ?;";
+
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, loginRequest.getEmail());
+
+            ResultSet resultSet = statement.executeQuery();
+            Argon2PasswordEncoder arg2SpringSecurity = getPasswordEncoder();
```

Comment:
No need to create this object if we have an empty result set... this cold be moved inside the while statement